File: NET/RestSharpTests/RestSharpTests/Base/ExtentReportListener.cs
using System;
using System.Reflection;
using AventStack.ExtentReports;
using AventStack.ExtentReports.Reporter;
using NUnit.Framework;
using NUnit.Framework.Interfaces;
using RestSharp;

namespace RestSharpTests.Base
{

    public class ExtentReportListener
    {
        public static AventStack.ExtentReports.ExtentReports Reports;
        public static ExtentTest extentTest;
        private static readonly string ResultPath = GetResultPath();

        private static string GetResultPath()
        {
            string path = Assembly.GetCallingAssembly().CodeBase;
            string actualPath = path.Substring(0, path.LastIndexOf("bin"));
            string projectPath = new Uri(actualPath).LocalPath;


            string resultPath = projectPath + "test-output\\";
            if (!Directory.Exists(resultPath))
            {
                Directory.CreateDirectory(resultPath);
            }
            return resultPath;
        }

        private string ReportLocation = $"{ResultPath}Report\\";



        public void SetUpReports(TestContext context)
        {

            var currentTime = DateTime.Now.ToString("yyyy-MM-dd hh-mm.ss");
            var sparkReporter = new ExtentSparkReporter($"{ReportLocation}ExtentReport_{currentTime}.html");
            Reports = new AventStack.ExtentReports.ExtentReports();
          
            Reports.AttachReporter(sparkReporter);
            Reports.AddSystemInfo("Application Name", "Testing");
            Reports.AddSystemInfo("Environment", Environment.GetEnvironmentVariable("env"));
            Console.WriteLine($"{ReportLocation}Reportlocation");
        }

        public void TestStarted()
        {
            extentTest = Reports.CreateTest(TestContext.CurrentContext.Test.MethodName);
            Console.WriteLine(TestContext.CurrentContext.Test.ClassName);
            Console.WriteLine(TestContext.CurrentContext.Test.MethodName);

        }

        public void TestFinished()
        {
            TestOutput();
            Reports.Flush();
        }

        public void TestOutput()
        {
            try
            {
                var status = TestContext.CurrentContext.Result.Outcome.Status;
                var stacktrace = string.IsNullOrEmpty(TestContext.CurrentContext.Result.StackTrace)
                  ? ""
                  : string.Format("{0}", TestContext.CurrentContext.Result.StackTrace);

                var errorMessage = TestContext.CurrentContext.Result.Message;
                Status logstatus;
                switch (status)
                {
                    case TestStatus.Failed:
                        logstatus = Status.Fail;
                        extentTest.Log(Status.Fail, "Test ended with " + Status.Fail + " – " + errorMessage);
                        extentTest.Log(Status.Fail, stacktrace);
                        break;
                    case TestStatus.Skipped:
                        logstatus = Status.Skip;
                        extentTest.Log(Status.Skip, "Test ended with " + Status.Skip);
                        break;
                    default:
                        logstatus = Status.Pass;
                        extentTest.Log(Status.Pass, "Test ended with " + Status.Pass);
                        extentTest.Log(Status.Pass, "Test steps finished for test case " + TestContext.CurrentContext.Test.Name);
                        
                        break;
                }
            }
            catch (Exception ex)
            {
                throw ex;
            }

        }

        public void LogMessage(Status st, string msg)
        {
            extentTest.Log(st, msg);
        }
    }
}

File: untitled/src/main/java/org/example/ExtentReportListener.java
package org.example;

import com.aventstack.extentreports.ExtentReports;
import com.aventstack.extentreports.ExtentTest;
import com.aventstack.extentreports.Status;
import com.aventstack.extentreports.reporter.ExtentSparkReporter;
import org.testng.ITestContext;
import org.testng.ITestListener;
import org.testng.ITestResult;
import java.io.File;
import java.text.SimpleDateFormat;
import java.util.Calendar;
public class ExtentReportListener implements ITestListener {
    public static ExtentReports reports;
    public static ExtentTest test;
    private static String resultPath = getResultPath();
    private static String getResultPath() {
        resultPath = "test";
        if (!new File(resultPath).isDirectory()) {
            new File(resultPath).mkdirs();
        }
        return resultPath;
    }
    String ReportLocation = "test-ouput/Report/" + resultPath + "/";
    public void onTestStart(ITestResult result) {
        test = reports.createTest(result.getMethod().getMethodName());
        System.out.println(result.getTestClass().getTestName());
        System.out.println(result.getMethod().getMethodName());
    }
    public void onTestSuccess(ITestResult result) {
        test.log(Status.PASS, "Test is passed");
    }
    public void onTestFailure(ITestResult result) {
        test.log(Status.FAIL, "Test is failed");
    }
    public void onTestSkipped(ITestResult result) {
        test.log(Status.SKIP, "Test is skipped");
    }
    public void onStart(ITestContext context) {
        String currentTime = new SimpleDateFormat("yyyy-MM-dd hh-mm.ss").format(Calendar.getInstance().getTime());
        ExtentSparkReporter sparkReporter = new ExtentSparkReporter(ReportLocation + "ExtentReport" + "_" + currentTime + ".html");
        reports = new ExtentReports();
        reports.attachReporter(sparkReporter);
        reports.setSystemInfo("Application Name", "Testing");
        reports.setSystemInfo("Environment", System.getProperty("env"));
        System.out.println(ReportLocation + "Reportlocation");
    }
    public void onFinish(ITestContext context) {
        reports.flush();
    }
}